crates/resolver/src/name_server/connection_provider.rs

```diff
@@ -382,14 +382,17 @@ pub mod tokio_runtime {
     use tokio::net::UdpSocket as TokioUdpSocket;
 
     /// A handle to the Tokio runtime
-    #[derive(Clone, Copy)]
-    pub struct TokioHandle;
+    #[derive(Clone, Default)]
+    pub struct TokioHandle {
+        join_set: std::sync::Arc<std::sync::Mutex<tokio::task::JoinSet<Result<(), ProtoError>>>>,
```

Comment:
Style nit: this gets a little ugly, can we just import `Arc`, `Mutex` and `JoinSet` here?

Reply:
Good call. Fixed.